{{cookiecutter.repostory_name}}/README.md

```diff
@@ -80,14 +71,25 @@ done
 EOT
 
 chmod +x hooks/post-receive
-./hooks/post-receive
+```
+
+```sh
+# locally
+git remote add production root@<server>:~/repos/{{ cookiecutter.repostory_name }}.git
+git push production master
```

Comment:
I'm not sure what you mean. If you're thinking about the first ever deploy, it will indeed fail, but it always does, with or without these proposed changes.

This diff makes it hard to see, but I just swapped the order from:
1. server: create bare repo
2. local: do first push
3. server: create hook
4. server: manually call hook - doesn't do anything, hangs because it waits on stdin, and if you input something it will call `deploy.sh` anyway, which will fail

to 
1. server: create bare repo
2. server: create hook
3. local: do first push

First deploy will always fail because `deploy.sh` needs `prepare-os.sh` and `setup-prod.sh` to have been run before, but this is impossible before you have a checkout, which is created by the hook.

I think we could add `prepare-os.sh` and `setup-prod.sh` calls to the hook before `deploy.sh`, then first deploy should work, and more importantly, if `prepare-os.sh` is changed, for example by a `cookiecutter` update, there would be no need to login and manually invoke it, or forget and then possibly get weird errors.

Reply:
the different being that after your changes since post-receive fail - will the files be actually added to the filesystem? I though in that case they won't be and there will be no way to move forward.

Calling prepare-os sounds fine to mine - my only worry is that it requires sudo - does it? post-receive cannot possibly be expected to require any interactive stuff happen (like asking for password)

Reply:
I have reverted the reorder since it can be confusing and instead just added `prepare-os.sh` to the hook and as a manual invocation.

Yes, `prepare-os.sh` requires sudo. The docs assume the git deployment will be done under the root user. I have added extra text to draw attention to this and what to do if it's not the case.

Reply:
we are not deploying under most of the time, but it is kinda true most of the time sudo is not interactive (doesn't ask for password) when we deploy it so it should be fine I guess